### PR TITLE
enh(drawio): poc multi tab editor

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
description in this PR:
https://github.com/centreon/centreon-map/pull/1109

### CODE REVIEW

**Comment by bdauria** [comment]
@Yassir-BenBOUBKER don't forget to get rid of the unused comments

**Comment by smau-centreon** [comment]
Is it possible to have a bit more context about this feature, please?

**Comment by bdauria** [comment]
@Yassir-BenBOUBKER You have a lot of unused comments that should be deleted

File: src/main/webapp/js/diagramly/Menus.js [hunk 1]
```diff
@@ -2472,29 +2472,37 @@
 					graph.stopEditing();
 				}
 				
-				var data = (urlParams['pages'] != '0' || (editorUi.pages != null && editorUi.pages.length > 1)) ?
-					editorUi.getFileData(true) : mxUtils.getXml(editorUi.editor.getGraphXml());
+				// var data = (urlParams['pages'] != '0' || (editorUi.pages != null && editorUi.pages.length > 1)) ?
+				// 	editorUi.getFileData(true) : mxUtils.getXml(editorUi.editor.getGraphXml());
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Menus.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Menus.js [hunk 2]
```diff
@@ -2472,29 +2472,37 @@
...
+					msg.xml = mxUtils.getXml(data);
+					msg.viewId = editorUi.currentPage.getViewId();
 					
 					if (exit)
 					{
 						msg.exit = true;
 					}
 					
 					data = JSON.stringify(msg);
+
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Menus.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 3]
```diff
@@ -194,7 +228,7 @@ SelectPage.prototype.execute = function()
 		page = this.ui.currentPage;
 	
 		// Switches the root cell and sets the view state
-		graph.model.prefix = Editor.guid() + '-';
+		// graph.model.prefix = Editor.guid() + '-';
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 4]
```diff
@@ -1133,14 +1172,14 @@ EditorUi.prototype.removePage = function(page)
...
-						mxResources.get('pageWithNumber', [1])));
-				}
+				// else if (this.pages.length <= 1)
+				// {
+				// 	Removes label with incorrect page number to force
+				// 	default page name which is OK for a single page
+				// 	next = this.insertPage();
+				// 	graph.model.execute(new RenamePage(this, next,
+				// 		mxResources.get('pageWithNumber', [1])));
+				// }
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 5]
```diff
@@ -1415,24 +1454,24 @@ EditorUi.prototype.updateTabContainer = function()
...
-		if (this.isPageInsertTabVisible())
-		{
-			insertTab = this.createPageInsertTab();
-			this.tabContainer.appendChild(insertTab);
-		}
+		// if (this.isPageInsertTabVisible())
+		// {
+		// 	insertTab = this.createPageInsertTab();
+		// 	this.tabContainer.appendChild(insertTab);
+		// }
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 6]
```diff
@@ -1415,24 +1454,24 @@ EditorUi.prototype.updateTabContainer = function()
 		this.tabContainer.appendChild(wrapper);
 		
 		// Adds floating menu with all pages and insert option
-		var menutab = this.createPageMenuTab();
-		this.tabContainer.appendChild(menutab);
+		// var menutab = this.createPageMenuTab();
+		// this.tabContainer.appendChild(menutab);
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 7]
```diff
@@ -1415,24 +1454,24 @@ EditorUi.prototype.updateTabContainer = function()
...
+		// }
 
 		if (wrapper.clientWidth > this.tabContainer.clientWidth - btnWidth)
 		{
 			if (insertTab != null)
 			{
 				insertTab.style.position = 'absolute';
 				insertTab.style.right = '0px';
-				wrapper.style.marginRight = '30px';
+				// wrapper.style.marginRight = '30px';
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 8]
```diff
@@ -1616,7 +1655,7 @@ EditorUi.prototype.createPageMenuTab = function(hoverEnabled, invert)
 				if (!invert)
 				{
 					menu.addSeparator(parent);
-					addInsert();
+					// addInsert();
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 9]
```diff
@@ -1650,8 +1689,8 @@ EditorUi.prototype.createPageMenuTab = function(hoverEnabled, invert)
 
 			if (invert)
 			{
-				menu.addSeparator(parent);
-				addInsert();
+				// menu.addSeparator(parent);
+				// addInsert();
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 10]
```diff
@@ -1739,11 +1778,11 @@ EditorUi.prototype.addTabListeners = function(page, tab)
...
-	mxEvent.addListener(tab, 'dblclick', mxUtils.bind(this, function(evt)
-	{
-		this.renamePage(page)
-		mxEvent.consume(evt);
-	}));
+	// mxEvent.addListener(tab, 'dblclick', mxUtils.bind(this, function(evt)
+	// {
+	// 	this.renamePage(page)
+	// 	mxEvent.consume(evt);
+	// }));
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 11]
```diff
@@ -1837,77 +1876,86 @@ EditorUi.prototype.createPageMenu = function(page, label)
...
 		var model = graph.model;
 
-		menu.addItem(mxResources.get('insert'), null, mxUtils.bind(this, function()
-		{
-			this.insertPage(null, mxUtils.indexOf(this.pages, page) + 1);
-		}), parent);
+		// menu.addItem(mxResources.get('insert'), null, mxUtils.bind(this, function()
+		// {
+		// 	this.insertPage(null, mxUtils.indexOf(this.pages, page) + 1);
+		// }), parent);
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]:
```suggestion
```

File: src/main/webapp/js/diagramly/Pages.js [hunk 12]
```diff
@@ -1837,77 +1876,86 @@ EditorUi.prototype.createPageMenu = function(page, label)
...
-			menu.addItem(mxResources.get('openInNewWindow'), null, mxUtils.bind(this, function()
-			{
-				this.editor.editAsNew(this.getFileData(true, null, null, null, true, true));
-			}), parent);
-		}
+		// 	menu.addItem(mxResources.get('openInNewWindow'), null, mxUtils.bind(this, function()
+		// 	{
+		// 		this.editor.editAsNew(this.getFileData(true, null, null, null, true, true));
+		// 	}), parent);
+		// }
```
Comment (jdelpierre) [inline src/main/webapp/js/diagramly/Pages.js]: remove comments

**Comment by Yassir-BenBOUBKER** [comment]
@jdelpierre @bdauria 
for the comments, i keep them
for easy access when we need theme
i will delete theme for now

File: src/main/webapp/js/grapheditor/EditorUi.js [hunk 13]
```diff
@@ -1514,20 +1522,20 @@ EditorUi.prototype.installShapePicker = function()
...
-				}), 30);
+				return;
+				// var pt = mxUtils.convertPoint(this.container, mxEvent.getClientX(evt), mxEvent.getClientY(evt));
+				// mxEvent.consume(evt);
+
+				// // Asynchronous to avoid direct insert after double tap
+				// window.setTimeout(mxUtils.bind(this, function()
+				// {
+				// 	ui.showShapePicker(pt.x, pt.y);
+				// }), 30);
```
Comment (jdelpierre) [inline src/main/webapp/js/grapheditor/EditorUi.js]:
```suggestion
```

File: src/main/webapp/js/grapheditor/EditorUi.js [hunk 14]
```diff
@@ -5151,7 +5159,7 @@ EditorUi.prototype.save = function(name)
 				}
 			}
 
-			this.editor.setModified(false);
+			// this.editor.setModified(false);
```
Comment (jdelpierre) [inline src/main/webapp/js/grapheditor/EditorUi.js]:
```suggestion
```

**Review by jdelpierre — APPROVED** [review]

File: etc/propgen/convert.js [hunk 15]
```diff
@@ -116,6 +116,10 @@ async function main()
               en: 'Create Map From Container',
               fr: "Créer une carte à partir d'un conteneur",
             },
+            editContent: {
+              en: 'Edit Content',
```
Comment (Thebarda) [inline etc/propgen/convert.js]:
```suggestion
              en: 'Edit content',
```

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**